programs/buck_version.py

```diff
@@ -34,12 +34,20 @@ def fileno(self):
 
 def is_git(dirpath):
     dot_git = os.path.join(dirpath, '.git')
-    return all([
-        os.path.exists(dot_git),
-        os.path.isdir(dot_git),
-        which('git'),
-        sys.platform != 'cygwin',
-    ])
+    if(which('git') and sys.platform != 'cygwin'):
+        if(os.path.exists(dot_git) and os.path.isdir(dot_git)):
+            return True
+        try:
+            output = check_output(
+                ['git', 'rev-parse', '--is-inside-work-tree'],
+                cwd=dirpath)
+            return output.strip() == 'true'
+        except:
```

Comment:
If you want to use exceptions here, change this block to only handle the exact error type that gets raised when the process exits with a nonzero code. Although I would prefer using `subprocess.run` instead and checking the `returncode` manually.